### PR TITLE
Bigquery contributions

### PR DESCRIPTION
**Pull request by rupertbates**
## What are you doing in this PR?
#2891 Added acquisition tracking to BigQuery for subscriptions and recurring contributions, this PR does the same for single contributions via the payment-api

[**Trello Card**](https://trello.com/c/SrukASHo/3536-acquisitions-data-pipeline-work-contributions-0)

## Accessibility test checklist
 - [ ] [Tested with screen reader](https://accessibility.gutools.co.uk/testing/web/screen-readers/)
 - [ ] [Navigable with keyboard](https://accessibility.gutools.co.uk/testing/web/keyboard-navigation/)
 - [ ] [Colour contrast passed](https://accessibility.gutools.co.uk/testing/web/colour-contrast/)

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -124,8 +124,8 @@ lazy val `support-payment-api` = (project in file("support-payment-api"))
     buildInfoPackage := "app",
     buildInfoOptions += BuildInfoOption.ToMap,
     libraryDependencies ++= commonDependencies
-  ).dependsOn(`support-models`, `support-internationalisation`, `acquisition-event-producer`)
```
Comment (rupertbates) [inline build.sbt:127]: acquisition-event-producer is brought in by support-models

File: support-config/src/main/scala/com/gu/support/config/AmazonPayConfig.scala [hunk 2]
```diff
@@ -5,7 +5,7 @@ import com.typesafe.config.Config
 case class AmazonPayConfig(
   sellerId: String,
   clientId: String
-) extends TouchpointConfig
```
Comment (rupertbates) [inline support-config/src/main/scala/com/gu/support/config/AmazonPayConfig.scala:8]: Removing the requirement for these config classes to extend `TouchpointConfig` means that payment-api doesn't need to depend on support-config which is not really relevant for it.

File: support-payment-api/src/main/scala/backend/AmazonPayBackend.scala [hunk 3]
```diff
@@ -124,6 +129,14 @@ class AmazonPayBackend(cloudWatchService: CloudWatchService,
     ophanService.submitAcquisition(acquisition)
       .bimap(BackendError.fromOphanError, _ => ())
 
+  private def submitAcquisitionToBigQuery(
+    acquisition: AmazonPayAcquisition,
+    contributionData: ContributionData
+  ): EitherT[Future, BackendError, Unit] =
+    EitherT.fromEither[Future](bigQueryService.tableInsertRow(
```
Comment (tomrf1) [inline support-payment-api/src/main/scala/backend/AmazonPayBackend.scala:136]: Any idea if we need to worry about this being a blocking call?
Currently we return a response to the client after the `trackContribution` tasks have finished, even though their result does not affect the response. Which is perhaps not ideal

Comment (rupertbates) [reply support-payment-api/src/main/scala/backend/AmazonPayBackend.scala]: This is a good point, I wasn't too worried about it in a lambda but on an EC2 box we probably need to be a bit more careful. I've wrapped the call in a blocking future as suggested here:
https://docs.scala-lang.org/overviews/core/futures.html#blocking-inside-a-future

Comment (tomrf1) [reply support-payment-api/src/main/scala/backend/AmazonPayBackend.scala]: thanks!

**Review by tomrf1 — APPROVED** [review]
Looks good!
I just tested a single contribution in CODE and there were no errors

**Pull request merged**